x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable_factory.tsx

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { StartServicesAccessor } from '@kbn/core/public';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import { KibanaThemeProvider } from '@kbn/react-kibana-context-theme';
+import { pick } from 'lodash';
+import { i18n } from '@kbn/i18n';
+
+import type { ReactEmbeddableFactory } from '@kbn/embeddable-plugin/public';
+import { EuiResizeObserver } from '@elastic/eui';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import useUnmount from 'react-use/lib/useUnmount';
+import moment from 'moment';
+import {
+  apiHasExecutionContext,
+  initializeTimeRange,
+  initializeTitles,
+  useBatchedPublishingSubjects,
+} from '@kbn/presentation-publishing';
+import { DatePickerContextProvider, type DatePickerDependencies } from '@kbn/ml-date-picker';
+import { UI_SETTINGS } from '@kbn/data-plugin/common';
+import type { MlJob } from '@elastic/elasticsearch/lib/api/types';
+import usePrevious from 'react-use/lib/usePrevious';
+import { throttle } from 'lodash';
+import { BehaviorSubject, Subscription } from 'rxjs';
+import { ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE } from '..';
+import type { MlPluginStart, MlStartDependencies } from '../../plugin';
+import type { SingleMetricViewerEmbeddableApi, SingleMetricViewerEmbeddableState } from '../types';
+import { initializeSingleMetricViewerControls } from './single_metric_viewer_controls_initializer';
+import { initializeSingleMetricViewerDataFetcher } from './single_metric_viewer_data_fetcher';
+import { TimeSeriesExplorerEmbeddableChart } from '../../application/timeseriesexplorer/timeseriesexplorer_embeddable_chart';
+import { APP_STATE_ACTION } from '../../application/timeseriesexplorer/timeseriesexplorer_constants';
+import { getServices } from './get_services';
+import { useReactEmbeddableExecutionContext } from '../common/use_embeddable_execution_context';
+import './_index.scss';
+
+const RESIZE_THROTTLE_TIME_MS = 500;
+const containerPadding = 10;
+const minElemAndChartDiff = 20;
+interface AppStateZoom {
+  from?: string;
+  to?: string;
+}
+
+const errorMessage = i18n.translate('xpack.ml.singleMetricViewerEmbeddable.errorMessage"', {
+  defaultMessage: 'Unable to load the ML single metric viewer data',
```

Comment:
Nit - `Unable to the load data for the single metric viewer chart` would make it consistent with the style for the other two anomaly embeddables.